app/lib/router/router.dart

```diff
@@ -325,9 +325,18 @@ List<RouteBase> makeRoutes(Ref ref) {
       pageBuilder: (context, state) {
         return NoTransitionPage(
           key: state.pageKey,
-          child: RoomPage(convo: state.extra as Convo),
+          child: const RoomPage(),
         );
       },
+      routes: <RouteBase>[
+        GoRoute(
+          parentNavigatorKey: rootNavKey,
+          path: 'profile',
+          name: Routes.chatProfile.name,
+          redirect: authGuardRedirect,
+          builder: (context, state) => const RoomProfilePage(),
+        ),
+      ],
```

Comment:
yeah, that isn't a sufficient fix. It still relies on the outer caller of the parameter to _have set_ the convo... Which isn't working in the specific case that this is supposed to fix: when we come from a notification. In that context, all we have is the roomId and the knowledge it is a chat room. Which _should_ be enough to route there (push notifications later won't have any other way to interact than push a stringy-route). Same will also be true if you linked to another room in a chat or had any other internal link to a room (like we allow in Pins or News-bulletins).

Generally all routes that take a parameter like spaceId or roomId should be self-sufficient in that they figure out how to load that thing - even just for the fact that if you routed there and the other didn't properly update, you'd be looking at the wrong data. Take a look at how spaces does it for a reference.

Reply:
I think the most probable (which is also ideal) way is to load ```Convo``` object through the future provider by feeding ```roomId```, essentially handling all required cases for data.

Reply:
Example like the ```chatProvider``` we have already set up to return convo via id.